scipy/sparse/linalg/tests/test_propack.py

```diff
@@ -0,0 +1,248 @@
+    assert_allclose(sigma1[:k], sigma2, rtol=tol, atol=tol)
+
+    # check that singular vectors are orthogonal
+    assert_orthogonal(u1, u2, rtol=tol, atol=tol)
+    assert_orthogonal(vt1.T, vt2.T, rtol=tol, atol=tol)
+
+
+@pytest.mark.parametrize('ctor', (np.array, csr_matrix, csc_matrix))
+@pytest.mark.parametrize('precision', _dtype_map.keys())
+@pytest.mark.parametrize('irl', (True, False))
+@pytest.mark.parametrize('which', ('LM', 'SM'))
+def test_svdp(ctor, precision, irl, which):
+    np.random.seed(0)
+    dtype = _dtype_map[precision]
+    n, m, k = 10, 20, 3
+    if which == 'SM' and not irl:
+        with assert_raises(ValueError):
+            check_svdp(n, m, ctor, dtype, k, irl, which)
+    else:
+        check_svdp(n, m, ctor, dtype, k, irl, which)
+
+
+def load_real(folder, precision, file='illc1850.coord'):
+    dtype = _dtype_map[precision]
+    path = os.path.join(folder, precision, 'Examples', file)
+    # Coordinate Text File
+    with open(path) as f:
+        m, n, nnz = (int(val) for val in f.readline().split())
+        coord = np.array([[float(val) for val in line.split()] for line in f])
+    i = coord[:, 0].astype(int) - 1
+    j = coord[:, 1].astype(int) - 1
+    data = coord[:, 2].astype(dtype)
+    A = coo_matrix((data, (i, j)))
+    return A
+
+
+def load_complex(folder, precision):
+    file = 'mhd1280b.cua'
+
+    dtype = _dtype_map[precision]
+    path = os.path.join(folder, precision, 'Examples', file)
+    with open(path, "r") as f:
+        contents = f.readlines()
+    file_metadata = contents[1].split()
+    matrix_metadata = contents[2].split()
+    datum_length = 15  # hard code rather than getting from contents[3]
+
+    n_header = 4
+    n_total, n_indptr, n_indices, n_data, _ = (int(n) for n in file_metadata)
+    m, n, nnz, _ = (int(n) for n in matrix_metadata[1:])
+
+    line_indptr = n_header
+    line_indices = line_indptr + n_indptr
+    line_data = line_indices + n_indices
+
+    def _concatenate_lines(lines):
+        return "".join([line.rstrip() for line in lines])
+
+    indptr = _concatenate_lines(contents[line_indptr:line_indices])
+    indptr = np.asarray([int(i) for i in indptr.split()])-1
+
+    indices = _concatenate_lines(contents[line_indices:line_data])
+    indices = np.asarray([int(i) for i in indices.split()])-1
+
+    data = _concatenate_lines(contents[line_data:])
+    data = np.asarray([float(data[i:i+datum_length])
+                       for i in range(0, len(data), datum_length)])
+    real, imag= data[::2], data[1::2]
+    data = real + imag*1.0j
+    data.astype(dtype)
+
+    return csc_matrix((data, indices, indptr), (m, n))
+
+
+def load_sigma(folder, precision="double", irl=False):
+    dtype = _dtype_map[precision]
+    s_name = "Sigma_IRL.ascii" if irl else "Sigma.ascii"
+    path = os.path.join(folder, precision, 'Examples', 'Output', s_name)
+    with open(path) as f:
+        data = np.array([dtype(line.split()[1]) for line in f], dtype=dtype)
```

Comment:
Apparently old numpy versions won't construct from string like this, I'll look into accomplishing this another way

Reply:
Most natural options seem:
* choose `float` or `complex` for the casting from string to number, depending on `dtype` or
* use `complex` for the casting, then use `astype(dtype=dtype)` to cast the array to the appropriate type. Unfortunately, this produces a warning, and that would need to be silenced.

Reply:
Most natural options seem:
* choose `float` or `complex` for the casting from string to number, depending on `dtype` or
* use `complex` for the casting, then use `astype(dtype=dtype)` to cast the array to the appropriate type. Unfortunately, this produces a warning, and that would need to be silenced.

Reply:
I ended up using the built-in Python `complex`/`float` constructors to handle the string -> number conversion and then converting the array to the correct `dtype` after that.  Could be less-efficient (I didn't notice a slowdown), but I think acceptable for test code